### ISSUE TITLE
Select multiple images for adding new cations

### ISSUE THREAD

**Issue by SilentD1** [toshiaki1729/stable-diffusion-webui-dataset-tag-editor#53]
I really would like to be able to select more than one image from the main window, and then hit "apply changes to selected imageS" when I have many with the same caption. I am currently going through large datasets and selecting each image to edit is not fun. Yes Im using this to create captions. NOT just to edit

**Comment by toshiaki1729** [comment]
Yes… I totally agree with you about that, it is really redundant and annoying to select images each by each.
But this extension is working on gradio app, and it is an simple tool to create simple GUI for using ML models. Yes, actually, not designed for managing files like this extension.
To manage images like the explorer.exe or modern cloud apps (like google drive), creating GUI from scratch with javascript  (like [openOutpaint extension](https://github.com/zero01101/openOutpaint-webUI-extension) does) is needed.

**Comment by toshiaki1729** [comment]
Sorry, forget about previous comment.
I will implement the feature.

**Comment by SilentD1** [comment]
Thanks for trying, but I dont think you understood. Now you changed it to all displayed, but I still cant select the specific images I want. Say I have 1000 images. They dont have any tags yet. I want to select 20, type a caption and apply that caption to those 20 photos

**Comment by SilentD1** [comment]
Thanks for trying, but I dont think you understood. Now you changed it to all displayed, but I still cant select the specific images I want. Say I have 1000 images. They dont have any tags yet. I want to select 20, type a caption and apply that caption to those 20 photos. then I move on to the next batch.

Having to select each photo to become displayed, and first THEN I can apply a caption to them all is even more time consuming. The "filter by selected" tab, adds an extra unnecessary step that can be avoided. Just let us select any photo in the main window, and do with them what we want. MULTI SELECT.

If that is not possible I suggest another option:

You already ALMOST have it.

The "Filter by selection" tab, has a button "add selection" You can use the ENTER button on it.
This allows for quick arrow keys selection for each image. So thats good. But why do the extra step of adding each image to selection before I can edit their tag.

If you allow "apply changes to selected image" to be triggered by ENTER key. I could use arrow keys to select each image very fast and skip the filter by selection tab entirely.

That one feature would save me 80% time "arrow key, enter, arrow key enter, arrow key, enter" done

**Comment by toshiaki1729** [comment]
I totally agree with you, it is redundant and annoying to select each by each.
But this extension is working on gradio and it is simple tool for making simple GUI for ML. In short, it is not designed to manage files like this extension.
To make GUI for managing files as explorer.exe or modern cloud apps (like google drive) does, it is needed to make GUI with javascript from scratch (like [openOutpaint extension](https://github.com/zero01101/openOutpaint-webUI-extension)).
And, unfortunately, I don't have any experience or knowledge on making GUI working on web browser.
I reopen this issue as a reminder, but please note that there is no plan to actively add this feature.

**Comment by toshiaki1729** [comment]
I totally agree with you. It is redundant and annoying to select each by each.
But this extension is working on gradio, which is a simple tool to make simple GUI for ML. In other words, it is not designed to manage files as this extension do.
To manage files as explorer.exe or modern cloud apps do, it is needed to implement GUI with javascript from scratch (like [openOutpaint extension](https://github.com/zero01101/openOutpaint-webUI-extension)).
However, unfortunately, I have no experience or knowledge to make GUI working on web browser.
I reopen this issue as a reminder, but please note that I have no plan to actively add the feature you expected.

**Comment by toshiaki1729** [comment]
> That one feature would save me 80% time "arrow key, enter, arrow key enter, arrow key, enter" done

This is an operation in need, required, must be.
gradio gallery has **NO** way to get which image is selected **NOR** do multiple selection, and, currently, this extension knows which image is selected by interpreting HTML shown on the web browser.

**Comment by toshiaki1729** [comment]
> If you allow "apply changes to selected image" to be triggered by ENTER key. I could use arrow keys to select each image very fast and skip the filter by selection tab entirely.

This could be good for solving this issue.
But for future, I will try to implement as custom keybord shortcut feature.

**Comment by SilentD1** [comment]
avesome, that would be greatly appreciated. I just spent 5 hours editing captions. and unfortunately there is no better way to manually create captions to my knowledge. Your extension is quite good. I especially like the negative filter options that hide images that I already tagged. great option.

**Comment by a-l-e-x-d-s-9** [comment]
It seems like the most natural workflow, to select multiple images from the preview and apply certain tags to them all at once.
Going one by one is a Sisyphean work even with a small number of images.

**Comment by noisefloordev** [comment]
I ran into this too.  For example, if you have an image set of a character with two outfits, and you want to tag each outfit correctly for training.  You want it to say "Add the tag?" for each image and just click "Yes" "No" "Yes" "Yes" "No".

This is the best way I've found.  This was discussed above, but it's bured in conversation, here are steps in case it helps anyone else:

- Select "Filter by selection"
- Click the first image in the grid
- If the image should have the tag, press Enter to add it to the list.
- Press Right to move to the next image.
- Repeat until you reach the end

When you're at the end, click "Apply selection filter".  Now you can use Batch Edit Captions to add the tag.

Takes a little getting used to, but it's not too bad.  It has some quirks (glitchy keyboard focus, etc.), but they're Gradio and AUTO111 problems, not this extension's.
